serverless-cljs-plugin/lumo/build.cljs

```diff
@@ -0,0 +1,195 @@
+  [zip-path cljs-lambda-opts]
+  (let [{:keys [name target-path source-paths resource-paths]}
+        (:project cljs-lambda-opts)
+
+        {:keys [options]}
+        (:compiler cljs-lambda-opts)]
+
+    ;; improve this with clojure.spec
+    ;; (assert (sequential? functions) (gstr/format "Functions should be a sequence, got %s, exiting." functions))
+    (assert (map? options) (gstr/format "Compiler options should be a map, got %s. Exiting." options))
+    (assert (seqable? source-paths) (gstr/format "Inputs should be a sequence, got %s. Exiting." source-paths))
+
+    (let [compiler-options (merge options (compiler-defaults target-path name))
+          ;; TODO generate an index.js suitable for AWS Lambda
+          ;; index-path (generate-index! target-path options functions))
+          ]
+
+      ;; compiling
+
+      (compile! source-paths compiler-options)
+
+      ;; the only async step is this
+      (zip! zip-path
+            {:dirs #{target-path "node_modules"}
+             ;; :files index-path
+             }
+            compiler-options))))
+
+(comment
+  (def cljs-lambda-opts {:project {:name "logpoc"
+                                   :source-paths ["src"]
+                                   :target-path "target"}
+                         :compiler {:options {:language-in   :ecmascript5
+                                              :optimizations :none
+                                              :source-map    false
+                                              :main          "logpoc.core"}}})
+  (.then (build! (process.cwd) project-path cljs-lambda-opts) println println))
+
+(defn- cli-opt-helper
+  [parser opt-map-or-err]
+  (if-let [option (.getopt parser)]
+    (if-not (gobj/get option "error")
+      (do
+        (.log js/console " " (gobj/get option "option") " " (gobj/get option "optarg" "-"))
+        (recur parser (assoc opt-map-or-err
+                             (-> option (gobj/get "option") keyword)
+                             (gobj/get option "optarg" "true"))))
+      (js/Error. (gstr/format "command line argument error found at %s" (gobj/get option "optopt"))))
+    opt-map-or-err))
+
+(def ^:private cli-opt-string
+  ["z:(zip-path)"])
+
+(defn cli-options
+  "Compute the option map or a js/Error."
+  [argv]
+  (if (seq argv)
+    (do (.log js/console "parsing cli options")
+        (cli-opt-helper (go/BasicParser. (str/join cli-opt-string) (clj->js argv) 0) {}))
+    ;; TODO Improve this error, print out basic usage
+    (js/Error. "command line arguments are required.")))
+
+(comment
+  (def argv ["-z" "path"])
+  (cli-options argv)
+  )
+
+(defn ^:export -main [& args]
+  (let [opts-or-err (cli-options *command-line-args*)
+        conf (read-conf!)]
+    (if-not (instance? js/Error opts-or-err)
+      (do (.info js/console "Building" (get-in conf [:project :name]))
+          (build! (:z opts-or-err) (read-conf!)))
```

Comment:
The options are read from `cljs-lambda.edn` but `read-conf!` can be augmented in order to handle reading from anywhere basically.